lib/Ziti.swift

```diff
@@ -402,33 +424,41 @@ import CZitiPrivate
         self.initCallback = initCallback
         self.postureChecks = postureChecks
         
-        nfOpts = ziti_options(config: nil,
-                                controller: ctrlPtr,
-                                tls:tls,
-                                disabled: false,
-                                config_types: ziti_all_configs,
-                                api_page_size: 25,
-                                refresh_interval: 15,
-                                metrics_type: EWMA_1m,
-                                router_keepalive: 5,
-                                pq_mac_cb: Ziti.onMacQuery,
-                                pq_os_cb:  Ziti.onOsQuery,
-                                pq_process_cb: Ziti.onProcessQuery,
-                                pq_domain_cb: Ziti.onDomainQuery,
-                                app_ctx: self.toVoidPtr(),
+        zitiOpts = ziti_options(config: nil,
+                              controller: ctrlPtr,
+                              tls:tls,
+                              disabled: false,
+                              config_types: ziti_all_configs,
+                              api_page_size: 25,
+                              refresh_interval: 15,
+                              metrics_type: EWMA_1m,
+                              router_keepalive: 5,
+                              pq_mac_cb: postureChecks?.macQuery != nil ? Ziti.onMacQuery : nil,
+                              pq_os_cb:  postureChecks?.osQuery != nil ?  Ziti.onOsQuery : nil,
+                              pq_process_cb: postureChecks?.processQuery != nil ? Ziti.onProcessQuery : nil,
+                              pq_domain_cb: postureChecks?.domainQuery != nil ? Ziti.onDomainQuery : nil,
+                              app_ctx: self.toVoidPtr(),
                               events: ZitiContextEvent.rawValue | ZitiRouterEvent.rawValue | ZitiServiceEvent.rawValue | ZitiMfaAuthEvent.rawValue | ZitiAPIEvent.rawValue,
-                                event_cb: Ziti.onEvent)
+                              event_cb: Ziti.onEvent)
+        
+        // ziti_instance required if being managed by ZitiTunnel
+        var zi:UnsafeMutablePointer<ziti_instance_s>?
+        if let zt = self.zitiTunnel {
+            zi = zt.createZitiInstance(id.id, &(zitiOpts!))
+        }
         
-        let initStatus = ziti_init_opts(&(nfOpts!), loop)
+        let initStatus = ziti_init_opts(&(zitiOpts!), loop)
         guard initStatus == Ziti.ZITI_OK else {
             let errStr = String(cString: ziti_errorstr(initStatus))
             log.error("unable to initialize Ziti, \(initStatus): \(errStr)", function:"start()")
             initCallback(ZitiError(errStr, errorCode: Int(initStatus)))
             return
         }
         
-        // set log level
-        ZitiLog.setLogLevel(.INFO)
```

Comment:
defaults to info i assume so this line was removed as redundant? just confirming it's intentionally removed

Reply:
yes - it's defaulted to INFO already, and this line would overwrite a user's attempt to set log level before initializing Ziti